### PR TITLE
added javadocs

### PR DESCRIPTION
**Pull request by siyopao**
https://scalar-labs.atlassian.net/browse/DLT-205

This will be 1/3 for `transaction` probably.

### CODE REVIEW

File: src/main/java/com/scalar/database/transaction/consensuscommit/CommitHandler.java [hunk 1]
```diff
@@ -18,21 +18,36 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** */
+/** Commit snapshots to {@link DistributedStorage} */
```
Comment (yito88) [inline src/main/java/com/scalar/database/transaction/consensuscommit/CommitHandler.java]: I think this class handles the commit process of a transaction, not just store updated values to the storage.

File: src/main/java/com/scalar/database/transaction/consensuscommit/CommitMutationComposer.java [hunk 2]
```diff
@@ -19,10 +19,19 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * Keep track of mutations that need to be committed to {@link
```
Comment (yito88) [inline src/main/java/com/scalar/database/transaction/consensuscommit/CommitMutationComposer.java]: I think it is confusing. These mutations change the state of records to commit state.

Comment (siyopao) [reply src/main/java/com/scalar/database/transaction/consensuscommit/CommitMutationComposer.java]: That is confusing! Thank you. Fixed.

File: src/main/java/com/scalar/database/transaction/consensuscommit/CommitHandler.java (hunk 1 above)
Comment (siyopao) [reply src/main/java/com/scalar/database/transaction/consensuscommit/CommitHandler.java]: Thank you! Fixed.

File: src/main/java/com/scalar/database/transaction/consensuscommit/CommitHandler.java [hunk 3]
```diff
@@ -18,21 +18,41 @@
...
     this.storage = checkNotNull(storage);
     this.coordinator = checkNotNull(coordinator);
     this.recovery = checkNotNull(recovery);
   }
 
+  /**
+   * Implements the commit phase of a transaction. That is, all the records that have been changed
+   * by the transaction (dirty objects) will be written to persistent storage.
+   *
+   * <p>The commit phase is performed in two stages: the prepare stage and the commit stage. If the
```
Comment (feeblefakie) [inline src/main/java/com/scalar/database/transaction/consensuscommit/CommitHandler.java]: No offence at all, but I think this explanation is too vague and seems not really correct.
What is the definition of prepare stage and commit stage here ? And what is the `phase` ?
The code has some methods called `prepareRecords`, `commitState` and `commitRecords` and what belongs to which stage/phase ?
In other words, what do you exactly mean by `If the commit stage finishes successfully` ?

File: src/main/java/com/scalar/database/transaction/consensuscommit/CommitHandler.java [hunk 4]
```diff
@@ -18,21 +18,41 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** */
+/** Handler for the commit phase of a transaction. */
```
Comment (feeblefakie) [inline src/main/java/com/scalar/database/transaction/consensuscommit/CommitHandler.java]: I think here is the place to explain how a transaction work in scalar db. Please carefully read the paper, understand it and come back.

**Pull request closed without merging**